### PR TITLE
Add CODEOWNERS for automated PR review assignment

### PR DESCRIPTION
**Pull request by turbaszek**
> Code owners are automatically requested for review when someone opens a pull request that modifies code that they own. 
https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

I think we may try this feature. Currently contributors are not able to request review (unless they ping a committer). 

I tried to create some basic rules but I'm happy to get any suggestions!

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Core
+/airflow/executors/ @kaxil @XD-DENG @ashb @turbaszek
+/airflow/jobs/ @kaxil @ashb @XD-DENG
+/airflow/models @kaxil @XD-DENG @ashb @turbaszek
```
Comment (kaxil) [inline .github/CODEOWNERS]:
```suggestion
/airflow/models @kaxil @XD-DENG @ashb @turbaszek

# DAG Serialization
/airflow/serialization @kaxil
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# WWW
+/airflow/www/ @ryanahamilton
+
+# Providers
+/airflow/providers/google @turbaszek @mik-laj
+
+# Dev tools
+/.github/workflows/ @potuk @ashb @kaxil
+breeze @potiuk
+/dev/ @potuk @ashb @kaxil
```
Comment (XD-DENG) [inline .github/CODEOWNERS]:
```suggestion
/dev/ @potiuk @ashb @kaxil
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Core
+/airflow/executors/ @kaxil @XD-DENG @ashb @turbaszek
+/airflow/jobs/ @kaxil @ashb @XD-DENG
+/airflow/models @kaxil @XD-DENG @ashb @turbaszek
```
Comment (XD-DENG) [inline .github/CODEOWNERS]:
```suggestion
/airflow/models/ @kaxil @XD-DENG @ashb @turbaszek
```
To keep consistent with other lines and the [sample syntax](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax)

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+# DAG Serialization
+/airflow/serialization @kaxil
```
Comment (XD-DENG) [inline .github/CODEOWNERS]:
```suggestion
/airflow/serialization/ @kaxil
```

To keep consistent with other lines and the [sample syntax](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax)

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+# API
+/airflow/api/ @mik-laj @kaxil
+/airflow/api_connexion/ @mik-laj @kaxil
+
+# WWW
+/airflow/www/ @ryanahamilton
+
+# Providers
+/airflow/providers/google @turbaszek @mik-laj
```
Comment (XD-DENG) [inline .github/CODEOWNERS]:
```suggestion
/airflow/providers/google/ @turbaszek @mik-laj
```

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# Core
+/airflow/executors/ @kaxil @XD-DENG @ashb @turbaszek
+/airflow/jobs/ @kaxil @ashb @XD-DENG
+/airflow/models @kaxil @XD-DENG @ashb @turbaszek
+
+# DAG Serialization
+/airflow/serialization @kaxil
+
+# Kubernetes
+/airflow/kubernetes @dimberman
```
Comment (XD-DENG) [inline .github/CODEOWNERS]:
```suggestion
/airflow/kubernetes/ @dimberman
```

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+/airflow/kubernetes_executor_templates @dimberman
```
Comment (XD-DENG) [inline .github/CODEOWNERS]:
```suggestion
/airflow/kubernetes_executor_templates/ @dimberman
```

**Review by XD-DENG — DISMISSED** [review]
- Fix incorrect GitHub ID/handler for Jarek
- Minor change to keep consistent (also to keep consistent with the [sample syntax](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax))

File: .github/CODEOWNERS [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+/airflow/api_connexion/ @mik-laj @kaxil
+
+# WWW
+/airflow/www/ @ryanahamilton
+
+# Providers
+/airflow/providers/google @turbaszek @mik-laj
+
+# Dev tools
+/.github/workflows/ @potuk @ashb @kaxil
```
Comment (turbaszek) [inline .github/CODEOWNERS]:
```suggestion
/.github/workflows/ @potiuk @ashb @kaxil
```

**Comment by turbaszek** [comment]
Thanks @XD-DENG for the comments! 👌

File: .github/CODEOWNERS [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+# Providers
+/airflow/providers/google/ @turbaszek @mik-laj
+
+# Dev tools
+/.github/workflows/ @potiuk @ashb @kaxil
+breeze @potiuk
+/dev/ @potiuk @ashb @kaxil
+/provider_packages/ @potiuk
+/scripts/ @potiuk
```
Comment (potiuk) [inline .github/CODEOWNERS]: Should we also add some of the providers - such as google/ amazon/ few others that have clear stewardship? I am recently discussing it with AWS people (I sent them link to the PR) and we could also invite the Composer team for google provider and probably we can easily find other people for other parts

I guess those people do not have to be committers to be in CODEOWNERS? Or do they?

Comment (potiuk) [reply .github/CODEOWNERS]: And I also asked them to join the discussion and share their questions at the devlist :)

Comment (turbaszek) [reply .github/CODEOWNERS]: > I guess those people do not have to be committers to be in CODEOWNERS? Or do they?

I'm afraid they have to be:
> The people you choose as code owners must have write permissions for the repository.

Comment (potiuk) [reply .github/CODEOWNERS]: It's a pity :(, This means that the AWS/Compose people should start working on their committer's status :D

File: .github/CODEOWNERS [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+# WWW
+/airflow/www/ @ryanahamilton
+
+# Providers
+/airflow/providers/google/ @turbaszek @mik-laj
+
+# Dev tools
+/.github/workflows/ @potiuk @ashb @kaxil
+breeze @potiuk
```
Comment (potiuk) [inline .github/CODEOWNERS:30]:
```suggestion
breeze @potiuk
breeze-complete @potiuk
Dockerfile @potiuk
Dockerfile.ci @potiuk
```

File: .github/CODEOWNERS [hunk 11]
```diff
@@ -0,0 +1,36 @@
+# Core
+/airflow/executors/ @kaxil @XD-DENG @ashb @turbaszek
+/airflow/jobs/ @kaxil @ashb @XD-DENG
+/airflow/models/ @kaxil @XD-DENG @ashb @turbaszek
+
+# DAG Serialization
+/airflow/serialization/ @kaxil
```
Comment (ashb) [inline .github/CODEOWNERS]:
```suggestion
/airflow/serialization/ @kaxil @ashb
```

File: .github/CODEOWNERS [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+# Kubernetes
+/airflow/kubernetes/ @dimberman
+/airflow/kubernetes_executor_templates/ @dimberman
+/airflow/executors/celery_kubernetes_executor.py @dimberman
+/chart/ @dimberman
```
Comment (ashb) [inline .github/CODEOWNERS]:
```suggestion
/chart/ @dimberman @ashb
```

File: .github/CODEOWNERS [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+
+# Docs
+/docs/ @mik-laj @kaxil @vikramkoka
+
+# API
+/airflow/api/ @mik-laj @kaxil
+/airflow/api_connexion/ @mik-laj @kaxil
+
+# WWW
+/airflow/www/ @ryanahamilton
```
Comment (ashb) [inline .github/CODEOWNERS]:
```suggestion
/airflow/www/ @ryanahamilton @ashb
```

File: .github/CODEOWNERS [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+# Dev tools
+/.github/workflows/ @potiuk @ashb @kaxil
+breeze @potiuk
+breeze-complete @potiuk
+Dockerfile @potiuk
+Dockerfile.ci @potiuk
+/dev/ @potiuk @ashb @kaxil
+/provider_packages/ @potiuk
+/scripts/ @potiuk
```
Comment (ashb) [inline .github/CODEOWNERS]:
```suggestion
/provider_packages/ @potiuk @ashb
/scripts/ @potiuk @ashb
```

File: .github/CODEOWNERS [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+
+# Providers
+/airflow/providers/google/ @turbaszek @mik-laj
+
+# Dev tools
+/.github/workflows/ @potiuk @ashb @kaxil
+breeze @potiuk
+breeze-complete @potiuk
+Dockerfile @potiuk
+Dockerfile.ci @potiuk
```
Comment (ashb) [inline .github/CODEOWNERS]:
```suggestion
Dockerfile @potiuk @ashb
Dockerfile.ci @potiuk @ashb
```

**Review by ashb — CHANGES_REQUESTED** [review]
Adding myself to a few areas.

**Review by ashb — APPROVED** [review]

**Review by ryanahamilton — APPROVED** [review]

**Comment by turbaszek** [comment]
Ok, let's merge it and see how it works 🚀

**Pull request merged**